docs/glossary.md

```diff
@@ -397,6 +411,10 @@ enforcement mechanism.
 The scripting language used to create Decred smart contracts is not [Turing
 complete](https://en.wikipedia.org/wiki/Turing_completeness).
 
+#### Soft Fork
+
+A change to consensus rules that is backwards compatible. Nodes running older versions without the new rules can still validate transactions and blocks created with the new rules. New transaction types, for instance, can often be added with soft-forking changes. However, only nodes using the new rules will be able to understand the new transaction type.
```

Comment:
I think really should properly call out a couple of important features:

A change to consensus rules that is backwards compatible.  **The backwards compatibility is achieved by tricking old nodes into believing they are faithfully and fully validating the transactions and blocks even though they are not.**  Nodes running older versions without the new rules can still **partially** validate transactions and blocks created with the new rules.  New transaction types, for instance, can often be added with soft-forking changes. However, only nodes using the new rules will be able to understand the new transaction type.

Reply:
Good point. Changed to:

>A change to consensus rules that is backwards compatible. Nodes running older versions without the new rules can still validate transactions and blocks created with the new rules. However, nodes running older versions can only partially validate transactions and blocks created using the new rules. For instance, if a new transaction type is added in a soft-fork, only nodes using the new rules will be able to understand the new transaction type.

Reply:
I really think it should call out the reality that is tricking old nodes into believing they are faithfully and fully validating the transactions and blocks even though they are not.  That is a super important distinction that I don't see being made here.

Reply:
I hesitate to add the words "tricking" and "faithfully" here, because while I agree with these characterizations, they are characterizations, not objective facts (what technical docs should generally stick to). Tricking implies deception, and code cannot be deceived, only the humans running it. And we can't strictly assume the humans running old code don't know what the deal is. My knowledge is limited here though, and open to counter argument. I just think we should make these arguments as objectively and dispassionately as possible. E.g. if there's another example, or negative consequence that a reader would reasonably interpret as "tricking", that would be good to add.

Reply:
I see your point, but it is an objective fact that old nodes believe they are fully validating the transactions and blocks even though they are not just in the same way it is an objective fact that an old node will reject blocks that are using rules that are the result of a hard fork.

I suppose you can leave out "tricking" and "faithfully" if you feel they're more in the characterization realm, although, I would still argue they are objectively true.  I would be happy to just specifically call out "Under a soft fork, old nodes believe they are fully validating the transactions and blocks even though they are not."

Reply:
Well, I could argue that software doesn't have beliefs systems either:)

However, is has been common since the invention of programming for coders to anthropomorphize their code, imagining code objects 'believing' and 'looking', etc. So I think this falls squarely into standard, objective terminology. How does this sound?

>A change to consensus rules that is backwards compatible. Nodes running older versions without the new rules can still validate transactions and blocks. However, nodes running older versions can only partially validate transactions and blocks created by nodes using the new rules. This is because old nodes believe they are fully validating transactions, however, they are not able to undstand transactions created using the new rules. For instance, if a new transaction type is added via soft fork, only nodes using the new rules will be able to understand the new transaction type.

Reply:
Sounds good.  I thought you had already made the change and approved the PR, but I'll do it again once you make it.

Reply:
Oops. That is my normal process. Committed and pushed new text.